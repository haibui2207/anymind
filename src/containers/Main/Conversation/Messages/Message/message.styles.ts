import { createUseStyles } from "react-jss";

import { rem } from "../../../../../utils/jss";
import COLORS from "../../../../../constants/COLORS";

export default createUseStyles({
  root: {
    padding: rem(16),
    overflow: "auto",
    display: "flex",
    boxSizing: "border-box",
    "&.reverse": {
      flexDirection: "row-reverse",
      "& $avatarWrapper": {
        marginRight: 0,
        marginLeft: rem(12),
      },
    },
  },
  avatarWrapper: {
    flex: "none",
    backgroundColor: COLORS.bgWhite,
    overflow: "hidden",
    borderRadius: "50%",
    width: rem(40),
    height: rem(40),
    position: "relative",
    marginRight: rem(12),
  },
  avatar: {
    width: "100%",
    position: "absolute",
    top: "50%",
    left: "50%",
    transform: "translate(-50%, -50%)",
  },
  userName: {
    marginBottom: 0,
    marginTop: rem(4),
    fontSize: rem(12),
    lineHeight: rem(16),
    whiteSpace: "nowrap",
    overflow: "hidden",
    fontWeight: "bold",
  },
  dateTime: {
    display: "inline-block",
    fontWeight: "normal",
    color: COLORS.textGray,
    fontSize: rem(10),
    marginLeft: rem(12),
  },
  content: {},
  message: {
    fontSize: rem(14),
    lineHeight: rem(16),
  },
});
