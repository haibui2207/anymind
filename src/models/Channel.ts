interface IChannel {
  id: string;
  name: string;
}

export default IChannel;
