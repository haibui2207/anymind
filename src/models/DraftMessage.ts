interface IDraftMessage {
  channelId: string;
  userId: string;
  text: string;
}

export default IDraftMessage;
