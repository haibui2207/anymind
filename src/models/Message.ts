interface IMessage {
  messageId: string;
  text: string;
  datetime: string;
  userId: string;
}

export default IMessage;
