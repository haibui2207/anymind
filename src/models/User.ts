interface IUser {
  id: string;
  name: string;
  avatar: string;
}

export default IUser;
