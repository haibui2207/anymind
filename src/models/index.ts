// @ts-nocheck
export { default as IUser } from "./User";
export { default as IMessage } from "./Message";
export { default as IChannel } from "./Channel";
export { default as IDraftMessage } from "./DraftMessage";
