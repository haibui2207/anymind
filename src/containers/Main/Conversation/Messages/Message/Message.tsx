import React, { memo, useCallback } from "react";
import classNames from "classnames";

import { IMessage, IUser } from "../../../../../models";
import { StaticHTML } from "../../../../../components";

import useStyles from "./message.styles";

interface IProps extends IMessage {
  user?: IUser;
  isSender: boolean;
}

const Message: React.FC<IProps> = ({ user, text, datetime, isSender }) => {
  const classes = useStyles();

  const formatDateTime = useCallback(
    (value): string => new Date(value).toLocaleTimeString(),
    []
  );

  return (
    <div className={classNames(classes.root, isSender && "reverse")}>
      <div className={classes.avatarWrapper}>
        <img
          alt={user?.name}
          src={user?.avatar}
          className={classes.avatar}
          data-test="message-user-avatar"
        />
      </div>
      <div className={classes.content}>
        <p className={classes.userName} data-test="message-user-name">
          {user?.name}
          <span className={classes.dateTime} data-test="message-date-time">
            {formatDateTime(datetime)}
          </span>
        </p>
        <div
          className={classNames("ck-content", classes.message)}
          data-test="message-text"
        >
          <StaticHTML stringHtml={text} />
        </div>
      </div>
    </div>
  );
};

export default memo(Message);
