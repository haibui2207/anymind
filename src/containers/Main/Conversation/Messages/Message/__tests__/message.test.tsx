import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import { IMessage, IUser } from "../../../../../../models";
import Message from "../Message";

interface IProps extends IMessage {
  user?: IUser;
  isSender: boolean;
}

describe("Testing Main > Conversation > Messages > Message", (): void => {
  let wrapper: any;
  const props: IProps = {
    messageId: "7619097256175680316",
    text: "Text from Sam",
    datetime: "2021-09-29T03:51:06.783Z",
    userId: "Sam",
    isSender: true,
    user: { id: "Sam", name: "Sam", avatar: "" },
  };

  beforeAll((): void => {
    wrapper = mount(<Message {...props} />);
  });

  it("should render match snapshot", (): void => {
    const wrapper = shallow(<Message {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should show user avatar", (): void => {
    expect(wrapper.find('[data-test="message-user-avatar"]')).toHaveLength(1);
  });

  it("should show user name", (): void => {
    expect(wrapper.find('[data-test="message-user-name"]')).toHaveLength(1);
  });
});
