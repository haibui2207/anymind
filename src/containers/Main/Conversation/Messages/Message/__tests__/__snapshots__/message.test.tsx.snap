// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Main > Conversation > Messages > Message should render match snapshot 1`] = `
<div
  className="root-0-2-1 reverse"
>
  <div
    className="avatarWrapper-0-2-2"
  >
    <img
      alt="Sam"
      className="avatar-0-2-3"
      data-test="message-user-avatar"
      src=""
    />
  </div>
  <div
    className="content-0-2-6"
  >
    <p
      className="userName-0-2-4"
      data-test="message-user-name"
    >
      Sam
      <span
        className="dateTime-0-2-5"
        data-test="message-date-time"
      >
        10:51:06 AM
      </span>
    </p>
    <div
      className="ck-content message-0-2-7"
      data-test="message-text"
    >
      <StaticHTML
        stringHtml="Text from Sam"
      />
    </div>
  </div>
</div>
`;
